packages/thirdweb/src/react/web/ui/prebuilt/Account/index.tsx

```diff
@@ -0,0 +1,310 @@
+"use client";
+
+import { useSuspenseQuery } from "@tanstack/react-query";
+import type { Address } from "abitype";
+import type React from "react";
+import { createContext, useContext } from "react";
+import type { Chain } from "../../../../../chains/types.js";
+import type { ThirdwebClient } from "../../../../../client/client.js";
+import { resolveAvatar } from "../../../../../extensions/ens/resolve-avatar.js";
+import {
+  type ResolveNameOptions,
+  resolveName,
+} from "../../../../../extensions/ens/resolve-name.js";
+import { getSocialProfiles } from "../../../../../social/profiles.js";
+import type { SocialProfile } from "../../../../../social/types.js";
+import { parseAvatarRecord } from "../../../../../utils/ens/avatar.js";
+import { getWalletBalance } from "../../../../../wallets/utils/getWalletBalance.js";
+
+type AccountProps = {
+  address: Address;
+  client: ThirdwebClient;
+};
+
+const AccountProviderContext = /* @__PURE__ */ createContext<
+  AccountProps | undefined
+>(undefined);
+
+export function Account(props: React.PropsWithChildren<AccountProps>) {
+  return (
+    <AccountProviderContext.Provider value={props}>
+      {props.children}
+    </AccountProviderContext.Provider>
+  );
+}
+
+/**
+ * @internal
+ */
+function useAccountContext() {
+  const ctx = useContext(AccountProviderContext);
+  if (!ctx) {
+    throw new Error(
+      "AccountProviderContext not found. Make sure you are using Account.Name, Account.Avatar, etc. inside an <Account /> component",
+    );
+  }
+  return ctx;
+}
+
+interface AccountBalanceProps
+  extends Omit<React.HTMLAttributes<HTMLSpanElement>, "children"> {
+  chain?: Chain;
+  tokenAddress?: string;
+  /**
+   * A function to format the balance's display value
+   */
+  formatFn?: (num: number) => number;
+}
+
+/**
+ *
+ * ### Basic usage
+ * ```tsx
+ * import { Account } from "thirdweb/react";
+ *
+ * <Account address="0x...">
+ *   <Account.Balance />
+ * </Account>
+ * ```
+ * Result:
+ * ```html
+ * <span>1.091435 ETH</span>
+ * ```
+ *
+ *
+ * ### Format the balance (round up, shorten etc.)
+ * The Account.Balance component accepts a `formatFn` which takes in a number and output a number
+ * The function is used to modify the display value of the wallet balance
+ *
+ * ```tsx
+ * const roundTo1Decimal(num: number):number => Math.round(num * 10) / 10;
+ *
+ * <Account.Balance formatFn={roundTo1Decimal} />
+ * ```
+ *
+ * Result:
+ * ```html
+ * <span>1.1 ETH</span>
+ * ```
+ */
+Account.Balance = ({
```

Comment:
Make sure to keep each component in its own file, and do the trick I did in the other PR.

Otherwise this will never show up in docs, also tree shaking doesn't work if you do it this way